internal/bgp/frr/config.go

```diff
@@ -345,13 +352,23 @@ func debouncer(body func(config *frrConfig),
 				if !ok { // the channel was closed
 					return
 				}
-				config = newCfg
+				if newCfg.useOld && config == nil {
+					continue // just ignore the event
+				}
+				if !newCfg.useOld {
+					config = newCfg.config
+				}
 				if !timerSet {
 					timeOut = time.After(reloadInterval)
 					timerSet = true
 				}
 			case <-timeOut:
-				body(config)
+				err := body(config)
+				if err != nil {
+					timeOut = time.After(failureRetryInterval)
+					timerSet = true
```

Comment:
does this mean that after a fail the next config update won't happen until the failureretryinterval finishes? shouldn't we let new updates override this timeout?

Reply:
That is correct, and I thought that it makes sense to go slow after a failure happened (think about the case of the slow container startup, after a failure we want to give it time to recover). 5 seconds is still a bearable timeout for applying configuration changes.